### PR TITLE
create edge server and pubsub emulator if needed

### PR DESCRIPTION
**Pull request by relud**

### CODE REVIEW

File: ingestion-edge/ingestion_edge/wsgi.py [hunk 1]
```diff
@@ -6,16 +6,20 @@
...
     """Main."""
     if __name__ == "__main__":
-        app.run(
-            host=environ.get("HOST", "0.0.0.0"), port=int(environ.get("PORT", 8000))
-        )
+        host = environ.get("HOST", "0.0.0.0")
+        port = int(environ.get("PORT", 8000))
+        sock = socket()
+        sock.bind((host, port))
+        print("Listening on port %s" % sock.getsockname()[1])
```
Comment (jklukas) [inline ingestion-edge/ingestion_edge/wsgi.py]: Should this be an INFO-level log message instead?

File: ingestion-edge/pubsub_emulator.py [hunk 2]
```diff
@@ -28,14 +28,33 @@ class PubsubEmulator(
...
         self.subscriptions: Dict[str, Subscription] = {}
         self.status_codes: Dict[str, grpc.StatusCode] = {}
         self.sleep = None
-        pubsub_pb2_grpc.add_PublisherServicer_to_server(self, server)
-        pubsub_pb2_grpc.add_SubscriberServicer_to_server(self, server)
+        self.port = port
+        self.max_workers = max_workers
+        self.new_server()
+
+    def new_server(self):
```
Comment (jklukas) [inline ingestion-edge/pubsub_emulator.py]: This name being a noun sounds like it should return some object representing the server. I'd prefer this be a verb like `create_server` or `initialize_server`.

Comment (jklukas) [reply ingestion-edge/pubsub_emulator.py]: And I now see that this is replacing a `create_server` that _did_ return a value 😆

File: ingestion-edge/pytest.ini [hunk 3]
```diff
@@ -8,3 +8,8 @@ filterwarnings =
     ignore::PendingDeprecationWarning:google.cloud.pubsub_v1.publisher.client:89
     ignore::PendingDeprecationWarning:google.cloud.pubsub_v1.subscriber.client:83
 docstyle_match = (?!test_)(?!conftest).*\.py$
+env =
+    D:HOST=localhost
+    D:PORT=0
+    D:ROUTE_TABLE=[["/submit/<suffix:path>","test",["POST"]]]
+    D:QUEUE_PATH=:memory:
```
Comment (jklukas) [inline ingestion-edge/pytest.ini]: This `D:` syntax for defaults feels a bit arcane. Can you add a comment linking to https://github.com/MobileDynasty/pytest-env#usage ?

File: ingestion-edge/pubsub_emulator.py (hunk 2 above)
Comment (relud) [reply ingestion-edge/pubsub_emulator.py]: I'll rename it back to create_server

**Comment by jklukas** [comment]
The beam integration test failed spuriously. Noted in https://github.com/mozilla/gcp-ingestion/issues/263 and kicked tests off again manually.

**Comment by jklukas** [comment]
It looks like ingestion-edge tests are consistently timing out due to 10 minutes of inactivity.

**Pull request merged**